extras/splat/splat-data.js

```diff
@@ -245,6 +246,129 @@ class SplatData {
             app.drawLines(debugLines, debugColor);
         }
     }
+
+    // compressed splats
+    get isCompressed() {
+        return this.elements.some(e => e.name === 'chunk') &&
+               ['packed_position', 'packed_rotation', 'packed_scale', 'packed_color'].every(name => this.getProp(name));
+    }
+
+    decompress() {
+        const members = ['x', 'y', 'z', 'f_dc_0', 'f_dc_1', 'f_dc_2', 'opacity', 'rot_0', 'rot_1', 'rot_2', 'rot_3', 'scale_0', 'scale_1', 'scale_2'];
+        const chunks = this.elements.find(e => e.name === 'chunk');
+        const vertices = this.vertexElement;
+
+        // allocate uncompressed data
+        const data = {};
+        members.forEach((name) => {
+            data[name] = new Float32Array(vertices.count);
+        });
+
+        const getChunkProp = (name) => {
+            return chunks.properties.find(p => p.name === name && p.storage)?.storage;
+        };
+
+        const min_x = getChunkProp('min_x');
+        const min_y = getChunkProp('min_y');
+        const min_z = getChunkProp('min_z');
+        const max_x = getChunkProp('max_x');
+        const max_y = getChunkProp('max_y');
+        const max_z = getChunkProp('max_z');
+        const min_scale_x = getChunkProp('min_scale_x');
+        const min_scale_y = getChunkProp('min_scale_y');
+        const min_scale_z = getChunkProp('min_scale_z');
+        const max_scale_x = getChunkProp('max_scale_x');
+        const max_scale_y = getChunkProp('max_scale_y');
+        const max_scale_z = getChunkProp('max_scale_z');
+
+        const position = this.getProp('packed_position');
+        const rotation = this.getProp('packed_rotation');
+        const scale = this.getProp('packed_scale');
+        const color = this.getProp('packed_color');
+
+        const unpackUnorm = (value, bits) => {
+            const t = (1 << bits) - 1;
+            return (value & t) / t;
+        };
+
+        const unpack111011 = (value) => {
+            return {
```

Comment:
I'd consider removal of the object creation in these. Accept Vec3/4 instead and fill in the result.